src/infra/HttpServiceImpl.ts

```diff
@@ -39,34 +43,61 @@ export class HttpServiceImpl implements HttpService {
     url: string,
     queryParams: QueryParams,
     body: QueryParams,
+    clientSdk: Record<string, string>,
     authToken?: string
   ): Promise<T> {
     const sanitizedBodyParams = sanitizeQueryParams(body);
-    return this.fetch(url, queryParams, {
+    return fetch(url, queryParams, {
       method: RequestMethods.POST,
       body: JSON.stringify(sanitizedBodyParams),
       mode: 'cors',
       ...(authToken && { credentials: 'include' }),
       headers: {
+        'Client-SDK': formatAsHttpHeader(clientSdk),
         'Content-Type': 'application/json',
         ...(authToken && { Authorization: `Bearer ${authToken}`}),
       }
     })
       .then(res => res.json());
   }
+}
 
-  /**
-   * Perform a fetch, using the polyfill if needed.
-   */
-  private fetch(
-    url: string,
-    queryParams: QueryParams,
-    reqInit: RequestInit
-  ): Promise<Response> {
-    const urlWithParams = addParamsToURL(url, queryParams);
-    if (typeof(window) !== 'undefined' && window.fetch) {
-      return window.fetch(urlWithParams, reqInit);
-    }
-    return crossFetch(urlWithParams, reqInit);
+/**
+ * Perform a fetch, using the polyfill if needed.
+ */
+function fetch(
+  url: string,
+  queryParams: QueryParams,
+  reqInit: RequestInit
+): Promise<Response> {
+  const urlWithParams = addParamsToURL(url, queryParams);
+  if (typeof(window) !== 'undefined' && window.fetch) {
+    return window.fetch(urlWithParams, reqInit);
   }
+  return crossFetch(urlWithParams, reqInit);
 }
+
+/**
+ * Converts the JSON representing the Client-SDK agents into the format of an HTTP header.
```

Comment:
instead of `the format of an HTTP header` I would maybe say `the expected HTTP header format`?
I don't think there's anything "wrong" with putting json in an HTTP header it's just not conventional (afaik, I'm no http expert)